### ISSUE TITLE
Supported source version 'RELEASE_11' from annotation processor 'org.eclipse.ice.tests.dev.annotations.processors.LoggingDataElementProcessor' less than -source '14'

### ISSUE THREAD

**Issue by jayjaybillings** [eclipse/ice#472]
I encounter this error - `Supported source version 'RELEASE_11' from annotation processor 'org.eclipse.ice.tests.dev.annotations.processors.LoggingDataElementProcessor' less than -source '14'` - when I try to build on win64 with Java 14.

It is caused by using `@SupportedSourceVersion(SourceVersion.Release_11)` on the annotation processors. 

Do we actually need this? Is there any reason why we would want to limit the latest version of JDK that we could use? I would propose that we remove these.

@dbluhm What do you think? If you agree I'll push the changes.

**Comment by dbluhm** [comment]
Without it, a warning is emitted on my version at least:

```
No SupportedSourceVersion annotation found on org.eclipse.ice.tests.dev.annotations.processors.LoggingDataElementProcessor, returning RELEASE_6.
Supported source version 'RELEASE_6' from annotation processor 'org.eclipse.ice.tests.dev.annotations.processors.LoggingDataElementProcessor' less than -source '11'
```
This seems problematic but I'm not sure how deeply it affects things. If it's just a warning that we can ignore, we could probably filter it out or ignore warnings altogether and just use `assertThat(compilation).succeeds()` in the tests that are failing but we do risk another warning that does actually need to be addressed getting missed.

**Comment by dbluhm** [comment]
Actually, thinking again, the solution in both cases (omitting it or keeping it in) would require those tests be loosened up a bit. Dropping the `@SupportedSourceVersion` from the main `DataElementProcessor` and running tests, there were no apparent negative side-effects but having it default to `RELEASE_6` feels odd if nothing else.

**Comment by dbluhm** [comment]
#473 Should hopefully solve this issue. Warnings will pop up during testing but they will no longer cause the tests to fail. I left the supported source versions at 11 since that is our "official" development version at the moment but with those changes, it shouldn't matter when running from Java 14.

**Comment by jayjaybillings** [comment]
Can you put that in a bug fix branch instead of your fork please?

**Comment by dbluhm** [comment]
https://github.com/eclipse/ice/tree/fix/logging-processor-version

**Comment by jayjaybillings** [comment]
It fails for me. I'll have to look at this more later.

```java
DataElementProcessorTest.testDocStringsPreserved:333 value of:
    compilation.generatedFile(/TestImplementation.java).contents()
expected to contain:
    * AND ANOTHER ON A NEW LINE.
but was:
    
    import java.io.Serializable;
    
    import org.slf4j.Logger;
    import org.slf4j.LoggerFactory;
    
    import com.fasterxml.jackson.annotation.JsonAutoDetect;
    import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
    import com.fasterxml.jackson.annotation.JsonCreator;
    import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
    import com.fasterxml.jackson.annotation.JsonProperty;
    import com.fasterxml.jackson.core.JsonProcessingException;
    import com.fasterxml.jackson.databind.ObjectMapper;
    
    import lombok.AccessLevel;
    import lombok.Data;
    import lombok.Getter;
    import lombok.NoArgsConstructor;
    import lombok.NonNull;
    import lombok.Setter;
    
    import org.eclipse.ice.data.JavascriptValidator;
    import java.util.UUID;
    
    /**
     * This is an implementation of Test that satisfies the dependencies of
     * the @DataElement Annotation and was auto-generated by the ICE Framework.
     */
    @Data
    @NoArgsConstructor
    @JsonAutoDetect(
    	fieldVisibility = Visibility.ANY,
    	getterVisibility = Visibility.NONE,
    	isGetterVisibility = Visibility.NONE,
    	setterVisibility = Visibility.NONE
    )
    @JsonIgnoreProperties(ignoreUnknown = true)
    public class TestImplementation implements Test, Serializable {
    
    	/**
    	 * Logging tool
    	 */
    	private static final Logger logger = LoggerFactory.getLogger(TestImplementation.class);
    
    	/**
    	 * The private UUID of this element. This field is left out of matches().
    	 */
    	@NonNull
    	@Getter(AccessLevel.NONE)
    	@Setter(AccessLevel.NONE)
    	protected 
UUID privateId = UUID.randomUUID();
    
    	/**
    	 * A unique identifier for this element.
    	 */
    	protected 
long id = 0L;
    
    	/**
    	 * A simple name for the data.
    	 */
    	@NonNull
    	protected 
String name = "name";
    
    	/**
    	 * A simple description of the data
    	 */
    	@NonNull
    	protected 
String description = "description";
    
    	/**
    	 * A comment that annotates the data in a meaningful way.
    	 */
    	@NonNull
    	protected 
String comment = "no comment";
    
    	/**
    	 * The context (a tag) in which the data should be considered.
    	 */
    	@NonNull
    	protected 
String context = "default";
    
    	/**
    	 * This value is true if the element should be regarded by the client as required.
    	 */
    	protected 
boolean required = false;
    
    	/**
    	 * This value is true if the element should be regarded as a secret by the client, such as for passwords.
    	 */
    	protected 
boolean secret = false;
    
    	/**
    	 * The validator used to check the correctness of the data.
    	 */
    	protected 
JavascriptValidator<Test> validator;
    
    	/**
    	 * A UNIQUE STRING IN THE DOC STRING.
    
     AND ANOTHER ON A NEW LINE.
    	 */
    	public 
int testInt;
    
    	/**
    	 * Get privateId.
    	 * @return UUID
    	 */
    	public UUID getUUID() {
    		return privateId;
    	}
    
    	/**
    	 * All args constructor for TestImplementation.
    	 *
    	 * Used in JSON Deserialization.
    	 * @param privateId {@code UUID} for field privateId
    	 * @param id {@code long} for field id
    	 * @param name {@code String} for field name
    	 * @param description {@code String} for field description
    	 * @param comment {@code String} for field comment
    	 * @param context {@code String} for field context
    	 * @param required {@code boolean} for field required
    	 * @param secret {@code boolean} for field secret
    	 * @param validator {@code JavascriptValidator<Test>} for field validator
    	 * @param testInt {@code int} for field testInt
    	 */
    	@JsonCreator
    	public TestImplementation(
    		@JsonProperty("privateId") @NonNull UUID privateId,
    		@JsonProperty("id") long id,
    		@JsonProperty("name") @NonNull String name,
    		@JsonProperty("description") @NonNull String description,
    		@JsonProperty("comment") @NonNull String comment,
    		@JsonProperty("context") @NonNull String context,
    		@JsonProperty("required") boolean required,
    		@JsonProperty("secret") boolean secret,
    		@JsonProperty("validator") JavascriptValidator<Test> validator,
    		@JsonProperty("testInt") int testInt
    	) {
    		this.privateId = privateId;
    		this.id = id;
    		this.name = name;
    		this.description = description;
    		this.comment = comment;
    		this.context = context;
    		this.required = required;
    		this.secret = secret;
    		this.validator = validator;
    		this.testInt = testInt;
    	}
    
    	/**
    	 * Copy constructor for TestImplementation.
    	 * @param other Instance of TestImplementation to copy
    	 */
    	public TestImplementation(Test other) {
    		copy(other);
    	}
    
    	/**
    	 * This operation clones the object. Note that it differs from the base class
    	 * implementation in that it will return null if it cannot create the clone to
    	 * promote fast failure. See {@link java.lang.Object#clone()};
    	 */
    	@Override
    	public Object clone() {
    		try {
    			// Call the copy constructor to create the clone.
    			return new TestImplementation(this);
    		} catch (Exception e) {
    			logger.error("Unable to clone TestImplementation!", e);
    			return null;
    		}
    	}
    
    	/**
    	 * Copy the contents of another element into this element.
    	 * @param element the element to copy
    	 */
    	@Override
    	public void copy(@NonNull Test element) {
    		this.privateId = element.getUUID();
    		this.id = element.getId();
    		this.name = element.getName();
    		this.description = element.getDescription();
    		this.comment = element.getComment();
    		this.context = element.getContext();
    		this.required = element.isRequired();
    		this.secret = element.isSecret();
    		this.validator = element.getValidator();
    		this.testInt = element.getTestInt();
    	}
    
    	/**
    	 * This function checks deep equality of DataElements to see if all members are
    	 * equal ("match") with the exception of fields with match set to false (such
    	 * as an automatically generated UUID). This is important for checking if two
    	 * objects were generated separately but are otherwise equal.
    	 *
    	 * @param o the other element to compare
    	 * @return true if all members of the element except excluded fields match
    	 *         this element.
    	 */
    	@Override
    	public boolean matches(Object o) {
    		boolean retval = false;
    
    		// Outer check for null comparisons
    		if (o != null) {
    			// Next check for shallow comparison
    			if (this != o) {
    				if (o instanceof TestImplementation) {
    					TestImplementation other = (TestImplementation) o;
    
    					// Separate boolean checks used to enable better catching
    					// by debuggers.
    
    					// Matching id
    					boolean idMatch =
    						this.id == other.id;
    
    					// Matching name
    					boolean nameMatch =
    						this.name.equals(other.name);
    
    					// Matching description
    					boolean descriptionMatch =
    						this.description.equals(other.description);
    
    					// Matching comment
    					boolean commentMatch =
    						this.comment.equals(other.comment);
    
    					// Matching context
    					boolean contextMatch =
    						this.context.equals(other.context);
    
    					// Matching required
    					boolean requiredMatch =
    						this.required == other.required;
    
    					// Matching secret
    					boolean secretMatch =
    						this.secret == other.secret;
    
    					// Matching validator
    					boolean validatorMatch =
    						this.validator == null ?
    							this.validator == other.validator :
    							this.validator.equals(other.validator);
    
    					// Matching testInt
    					boolean testIntMatch =
    						this.testInt == other.testInt;
    
    					retval =
    						idMatch && 
nameMatch && 
descriptionMatch &&

    						commentMatch && 
contextMatch && 
requiredMatch &&

    						secretMatch && 
validatorMatch && 
testIntMatch
;
    				}
    			} else {
    				// This should be true if they are the same because the deep comparison is
    				// performed otherwise.
    				retval = true;
    			}
    		}
    		return retval;
    	}
    
    	/**
    	 * This operation serializes the data element to a string in verified JSON.
    	 *
    	 * @return a JSON string describing the element
    	 */
    	@Override
    	public String toJson() {
    		String value = null;
    		// Convert to json using Jackson
    		ObjectMapper mapper = new ObjectMapper();
    		try {
    			value = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(this);
    		} catch (JsonProcessingException e) {
    			logger.error("Unable to write DataElement to string!", e);
    		}
    
    		return value;
    	}
    
    	/**
    	 * This operation deserializes a valid JSON string and tries to load it into the
    	 * object.
    	 *
    	 * @param jsonDataElement the contents of this data element as JSON
    	 */
    	@Override
    	public Test fromJson(final String jsonDataElement) {
    		// Load the data from the string with Jackson.
    		ObjectMapper mapper = new ObjectMapper();
    
    		try {
    			copy(mapper.readValue(jsonDataElement, TestImplementation.class));
    		} catch (JsonProcessingException e) {
    			logger.error("Unable to read DataElement from string!", e);
    		}
    
    		return this;
    	}
    }
compilation was:
    compilation of [/C:/Users/jayjaybillings/git/ice/org.eclipse.ice.dev/org.eclipse.ice.dev.annotations/target/test-classes/input/DataElement/Single.java] using annotation processors [lombok.launch.AnnotationProcessorHider$AnnotationProcessor@2728e166, org.eclipse.ice.dev.annotations.processors.DataElementProcessor@1d81dece]
```

**Comment by dbluhm** [comment]
Looks like a different error... Interesting, the doc comment formatting appears to not be behaving the same on Java 11 + Fedora as it is for you on Java 14 + Windows. I'll see if running it on Java 14 on Fedora has the same effect.

**Comment by dbluhm** [comment]
Could not reproduce on Java 14 on Fedora. On closer inspection, I think it must be a newline issue. Java on windows will interpret `"\n"` as the platform specific newline separator. As a fix, I've added the velocity escape tool to our velocity usage that should hopefully take care of that. Pushed those changes to the PR/bug-fix branch.

**Comment by jayjaybillings** [comment]
That fixed it.

So, do we want to go with removing the annotation or should we update the source version to 14? I guess the con would be that client devs would need Java 14. The pros would be that it wouldn't assume Java 6 and print that warning message.

**Comment by dbluhm** [comment]
Since we'll end up with warnings being omitted either way on 14 whether we remove it or leave it as 11, I just left it as 11. The main concern I have with changing it to 14 is that it will no longer work on 11 which is the release everything is currently set to target so we'd have to update all of those as well. And with that, everyone currently working on it would have to update their environments, too. So balancing the cost of updating everything right now with having warnings on 14, I'm leaning towards leaving it on 11 unless there are other more compelling reasons to update.

**Comment by jayjaybillings** [comment]
Okay. 

Fixed with the merger of #473.
